onnxruntime/core/providers/cuda/cuda_execution_provider.cc

```diff
@@ -70,7 +70,7 @@ CUDAExecutionProvider::CUDAExecutionProvider(const CUDAExecutionProviderInfo& in
   InsertAllocator(CreateAllocator(default_allocator_info, device_id_));
 
   DeviceAllocatorRegistrationInfo pinned_allocator_info(
-      {OrtMemTypeCPUOutput, [](int) { return std::make_unique<CUDAPinnedAllocator>(0, CUDA_PINNED); }, std::numeric_limits<size_t>::max()});
+      {OrtMemTypeCPUOutput, [](int id) { return std::make_unique<CUDAPinnedAllocator>(id, CUDA_PINNED); }, std::numeric_limits<size_t>::max()});
```

Comment:
Pinned allocator is in CPU, so no device id needed?

Reply:
The major reason is CudaAsyncBuffer need a device_id for CUDA allocator. There's more place need to be fixed. Will update later.

---
In reply to: [321858267](https://github.com/microsoft/onnxruntime/pull/1771#discussion_r321858267) [](ancestors = 321858267)